container/broker/lxd-broker.go

```diff
@@ -179,20 +178,9 @@ func (broker *lxdBroker) AllRunningInstances(ctx context.ProviderCallContext) (r
 	return broker.manager.ListContainers()
 }
 
-// MaintainInstance ensures the container's host has the required iptables and
-// routing rules to make the container visible to both the host and other
-// machines on the same subnet.
-func (broker *lxdBroker) MaintainInstance(ctx context.ProviderCallContext, args environs.StartInstanceParams) error {
-	machineID := args.InstanceConfig.MachineId
-
-	// There's no InterfaceInfo we expect to get below.
-	_, err := prepareOrGetContainerInterfaceInfo(
-		broker.api,
-		machineID,
-		false, // maintain, do not allocate.
-		lxdLogger,
-	)
-	return err
+// MaintainInstance is a no-op.
+func (broker *lxdBroker) MaintainInstance(_ context.ProviderCallContext, _ environs.StartInstanceParams) error {
```

Comment:
I think this means that `MaintainInstance` is always a no-op and could be removed from the interface and implementations altogether.